### PR TITLE
[CBRD-23609] Send messages to the system logger when there is insufficient space in operating system device.

### PR DESCRIPTION
**Pull request by hornetmj**
http://jira.cubrid.org/browse/CBRD-23609

- Failover with insufficient disk space does not leave enough logs.
- Therefore, it is too difficult for the DBA to figure out why failover occurred.
- Allows the system logger to record the information in case of problems with insufficient disk space.

### CODE REVIEW

File: src/transaction/log_manager.c [hunk 1]
```diff
@@ -575,6 +583,37 @@ log_get_eof_lsa (void)
...
+ * NOTE:
+ */
+void
+log_get_num_free_block (INT64 * num_free_block)
+{
+#if defined(SOLARIS)
+  struct statvfs buf;
+#else /* SOLARIS */
+  struct statfs buf;
+#endif /* SOLARIS */
```
Comment (applys76) [inline src/transaction/log_manager.c]: is this comment right?

File: src/transaction/log_manager.c [hunk 2]
```diff
@@ -575,6 +583,37 @@ log_get_eof_lsa (void)
...
+  memset (&buf, 0, sizeof (buf));
+  *num_free_block = 0;
+
+#if defined(SOLARIS)
+  if (statvfs (log_Path, &buf) == 0)
+#elif defined(AIX)
+  if (statfs ((char *) log_Path, &buf) == 0)
+#else /* AIX */
+  if (statfs (log_Path, &buf) == 0)
+#endif /* AIX */
```
Comment (applys76) [inline src/transaction/log_manager.c]: is this comment right?

File: src/transaction/log_manager.c [hunk 3]
```diff
@@ -575,6 +583,37 @@ log_get_eof_lsa (void)
...
+
+  memset (&buf, 0, sizeof (buf));
+  *num_free_block = 0;
+
+#if defined(SOLARIS)
+  if (statvfs (log_Path, &buf) == 0)
+#elif defined(AIX)
+  if (statfs ((char *) log_Path, &buf) == 0)
+#else /* AIX */
+  if (statfs (log_Path, &buf) == 0)
```
Comment (applys76) [inline src/transaction/log_manager.c]: Isn't it more accurate to judge by narrowing the error code to ENOMEM when performing statfs and failing?

File: src/transaction/log_manager.c (hunk 1 above)
Comment (hornetmj) [reply src/transaction/log_manager.c]: I referenced the file_io.c and log_manager.c

File: src/transaction/log_manager.c (hunk 2 above)
Comment (hornetmj) [reply src/transaction/log_manager.c]: I referenced the file_io.c and log_manager.c

File: src/transaction/log_manager.c (hunk 3 above)
Comment (hornetmj) [reply src/transaction/log_manager.c]: How about leaving syslog in this function if errno is ENOMEM or EIO?

Comment (applys76) [reply src/transaction/log_manager.c]: It's not bad. but the first idea was to let the cub_master know the information and cub_master leave the info to the log so finally make the 'admin' know the info from only logs by cub_master, isn't it?

File: src/transaction/log_manager.c (hunk 1 above)
Comment (applys76) [reply src/transaction/log_manager.c]: I got it but please confirm it's the right comment.

File: src/transaction/log_manager.c (hunk 2 above)
Comment (applys76) [reply src/transaction/log_manager.c]: I got it but please confirm it's the right comment.

File: src/transaction/log_manager.c (hunk 1 above)
Comment (hornetmj) [reply src/transaction/log_manager.c]: fix it.

File: src/transaction/log_manager.c (hunk 2 above)
Comment (hornetmj) [reply src/transaction/log_manager.c]: fix it.

File: src/transaction/log_manager.c (hunk 3 above)
Comment (hornetmj) [reply src/transaction/log_manager.c]: The cub_master may not be able to write an error log due to same problem of a file system or disk of a cub _server if errno is ENOMEM or EIO.

**Pull request merged**